persist/nft.go

```diff
@@ -150,3 +152,83 @@ func NftUpdateById(pIDstr DbId, updatedNft *Nft, pCtx context.Context, pRuntime
 	return mp.Update(pCtx, bson.M{"_id": pIDstr}, updatedNft)
 
 }
+
+//-------------------------------------------------------------
+
+func NftBulkUpsertOrRemove(walletAddress string, nfts []*Nft, pCtx context.Context, pRuntime *runtime.Runtime) error {
+
+	mp := NewMongoStorage(0, nftColName, pRuntime)
+
+	// UPSERT
+	// --------------------------------------------------------
+	weWantToUpsertHere := true
+
+	upsertModels := make([]mongo.WriteModel, len(nfts))
+
+	for i, v := range nfts {
+
+		if v.OpenSeaIDstr == "" {
+			return errors.New("open sea id required for each nft")
+		}
+
+		now := float64(time.Now().UnixNano()) / 1000000000.0
+
+		// TODO last updated
+
+		upsertModels[i] = mongo.UpdateOneModel{
+			Upsert: &weWantToUpsertHere,
+			Filter: bson.M{"owner_address": walletAddress, "opensea_id": v.OpenSeaIDstr},
+			Update: bson.M{
+				"$setOnInsert": bson.M{"_id": generateId(now), "created_at": now},
+				"$set":         v,
+			},
+		}
+	}
+
+	if _, err := mp.collection.BulkWrite(pCtx, upsertModels); err != nil {
+		return err
+	}
+
+	// FIND DIFFERENCE AND DELETE OUTLIERS
+	// -------------------------------------------------------
+	opts := &options.FindOptions{}
+	if deadline, ok := pCtx.Deadline(); ok {
+		dur := time.Until(deadline)
+		opts.MaxTime = &dur
+	}
+
+	allNfts := []*Nft{}
+	if err := mp.Find(pCtx, bson.M{"owner_address": walletAddress}, allNfts, opts); err != nil {
+		return err
+	}
+
+	if len(allNfts) > len(nfts) {
+		currOpenseaIds := map[string]bool{}
```

Comment:
Nit, non-blocking: consistent capitalization of OpenSea
```suggestion
		currOpenSeaIds := map[string]bool{}
```